NTBS-2119 Use responsive tables

benjimarshall opened the pull request:
Use the new [NHS responsive tables](https://service-manual.nhs.uk/design-system/components/table#3-or-more-column-table) added in v4 for tables with more than 3 columns, as recommended by the design system.

## Description
* Fix table captions from NHS v4 update: captions are no longer hidden in NHS v4. We were basically using them as aria-labels, so switch them to be aria-labels. 
* Use responsive tables for tables with more than 3 columns.
* Tweak the table padding CSS to make the table row hover highlighting look nice for narrow-viewport tables.
* Add an updated version of the NHS frontend fork, with responsive table templating added.

## Checklist:
- [x] Automated tests are passing locally.
- [x] If changing content for notification overview, confirmed renders okay for print in Chrome and IE
### Accessibility testing
Features with UI components should consider items on this list
- [x] Test functionality without javascript
- [x] Test in small window (imitating small screen)
- [x] Check the feature looks and works correctly in IE11
- [x] Zoom page to 400% - content still visible?
- [x] Test feature works with keyboard only operation
- [x] Test with one screen reader (e.g. [NVDA](https://www.nvaccess.org/): see [getting started](https://webaim.org/articles/nvda/)) - logical flow, no unexpected content. 
- [x] Passes automated checker (e.g. [WAVE](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh))

## Screenshots
<details>
  <summary>Dashboard page</summary>

### Dashboard page: wide
![wide_dashboard](https://user-images.githubusercontent.com/3650110/109475519-b9abcc80-7a6d-11eb-8e1b-db6d68cd341b.png)
### Dashboard page: narrow
![narrow_dashboard_1](https://user-images.githubusercontent.com/3650110/109475518-b9133600-7a6d-11eb-9616-5a207bc6628f.png)
![narrow_dashboard_2](https://user-images.githubusercontent.com/3650110/109475513-b9133600-7a6d-11eb-8a7d-83715309e617.png)
</details>

<details>
  <summary>Test results in overview</summary>

### Test results in overview: wide
![test_results_wide](https://user-images.githubusercontent.com/3650110/109475510-b87a9f80-7a6d-11eb-8224-4b7806e743eb.png)  
### Test results in overview: narrow
![test_results_narrow](https://user-images.githubusercontent.com/3650110/109475509-b7e20900-7a6d-11eb-84b8-f2e165f78a10.png)
</details>

<details>
  <summary>Test results in edit page</summary>

### Test results in edit page: wide
![test_results_more_details_wide](https://user-images.githubusercontent.com/3650110/109475505-b6b0dc00-7a6d-11eb-9e1a-a570c0267636.png)
### Test results in edit page: narrow
![test_results_more_details_narrow](https://user-images.githubusercontent.com/3650110/109475504-b6184580-7a6d-11eb-8fd0-a33301f56ada.png)
</details>

<details>
  <summary>Treatment events in overview</summary>

### Treatment events in overview: wide
![treatment_event_overview_wide](https://user-images.githubusercontent.com/3650110/109475512-b87a9f80-7a6d-11eb-8e1c-f55f53f98a5c.png)
### Treatment events in overview: narrow
![treatment_event_overview_narrow](https://user-images.githubusercontent.com/3650110/109475511-b87a9f80-7a6d-11eb-9aa5-a53ff018b9c0.png)
</details>

<details>
  <summary>Treatment events edit page</summary>

### Treatment events in edit page: wide
![treatment_event_more_details_wide](https://user-images.githubusercontent.com/3650110/109475508-b7e20900-7a6d-11eb-8c97-a13d49c1fccf.png)
### Treatment events in edit page: wide
![treatment_event_more_details_narrow](https://user-images.githubusercontent.com/3650110/109475506-b7497280-7a6d-11eb-9024-1540aeb109bc.png)
</details>

JoshCadney commented on ntbs-service/wwwroot/css/table.scss:
```diff
@@ -10,7 +10,23 @@
+    padding-left: 8px;
+}
+.nhsuk-table__header:last-child, .nhsuk-table__cell:last-child {
+    padding-right: 8px !important;
```
Why do we need to use `important` here? What rule are we competing against? Is there no other way to increase our specificity?

If you do need to use `important` then at a minimum I think you should add a comment explaining why.

benjimarshall replied:
I've had another look at it and found a way of increasing the specificity of the rules. We were competing against a style from the NHS library, but I've found a way of sorting it without using `!important`

JoshCadney reviewed (approved):
Looks good :-)

benjimarshall commented:
@JoshCadney I've rebased onto master, which didn't cause too many issues. Whilst doing some checking for the rebase I spotted a minor issue with table widths on the treatment event edit page, so I've just fixed that
